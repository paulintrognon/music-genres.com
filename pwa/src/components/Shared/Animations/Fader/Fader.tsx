import { animated, useTransition } from 'react-spring'

interface Props {
  isVisible: boolean
}
const Fader: React.FC<Props> = ({ isVisible, children }) => {
  const transition = useTransition(Boolean(isVisible), null, {
    from: { opacity: '0' },
    enter: { opacity: '1' },
    leave: { opacity: '0' },
    config: { duration: 200 },
  })

  return (
    <>
      {transition.map(
        ({ item, key, props }) =>
          item && (
            <animated.div style={props} key={key}>
              {children}
            </animated.div>
          )
      )}
    </>
  )
}
export default Fader
