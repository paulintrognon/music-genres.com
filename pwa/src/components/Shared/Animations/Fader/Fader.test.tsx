import { shallow } from 'enzyme'
import Fader from './Fader'

it('should return nothing if visible is false', () => {
  const wrapper = shallow(
    <Fader isVisible={false}>
      <p>Children</p>
    </Fader>
  )
  expect(wrapper).toMatchSnapshot()
})
