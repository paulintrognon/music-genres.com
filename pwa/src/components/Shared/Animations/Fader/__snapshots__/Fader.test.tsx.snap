// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return nothing if visible is false 1`] = `<Fragment />`;
